src/script/lua_api/l_internal.h

```diff
@@ -29,7 +29,20 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include "common/c_internal.h"
 
 #define luamethod(class, name) {#name, class::l_##name}
-#define luamethod_aliased(class, name, alias) {#name, class::l_##name}, {#alias, class::l_##name}
+
+#define luamethod_dep(class, good, bad)                                     \
+		{#bad, [](lua_State *L) -> int {                                    \
+			return l_deprecated_function(L, #bad, #good, &class::l_##good); \
+		}}
+
+#define luamethod_aliased(class, name, alias) \
+		{#name, class::l_##name},             \
+		luamethod_dep(class, name, alias)
+
+#define luamethod_aliased2(class, name, alias1, alias2) \
+		luamethod_aliased(class, name, alias1),         \
+		luamethod_dep(class, name, alias2)
```

Comment:
I don't think this is better than just manually inserting another entry for `luamethod_dep(...)`